bam_sort.c

```diff
@@ -201,6 +201,50 @@ static void pretty_header(char** text_in_out, int32_t text_len)
     *text_in_out = output;
 }
 
+static char* regex_escape(const char* input)
+{
+    size_t new_size = strlen(input) + 1;
+    char* output = (char*)malloc((2*new_size)+1);
+    if (output == NULL) {
+        perror("[regex_escape] Error allocating memory for regex escape buffer");
+        exit(1);
+    }
+    char* output_iter = output;
+    const char* input_iter = input;
+
+    while (*input_iter != '\0') {
+        switch (*input_iter) {
+            case '.':
+            case '[':
+            case '{':
+            case '}':
+            case '(':
+            case ')':
+            case '\\':
+            case '*':
+            case '+':
+            case '?':
+            case '|':
+            case '^':
+            case '$':
```

Comment:
On 17/08/15 13:38, John Marshall wrote:
>
> Why no |']'| (and yet the other flavours of closing brackets are there)?
>
Quizzed Martin about this and confirmed that opening ] didn't cause a 
problem.



-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.